mitm_service/tunneler/tunneler.py

```diff
@@ -50,25 +50,42 @@ def __init__(self, target_mac, gateway_mac, my_mac, target_ip, interface):
         self._raw_sock = _create_raw_ip_socket(interface)
         self._should_forward = False
         self._forward_thread = None
-        self._disruption_rules = {}
+        self._filters_by_layer = {}
         self.target_ip = inet_aton(target_ip)
-        for layer_num in type(self).DISRUPTION_CAPABLE_LAYERS:
-            self._disruption_rules[layer_num] = lambda header, payload: (header, payload)
+        for layer_num in type(self).FILTER_CAPABLE_LAYERS:
+            self._filters_by_layer[layer_num] = []
 
-    def add_disruption_rule(self, layer, rule):
+    def add_filter_to_layer(self, layer, resolution_index, protocol_filter):
         """
-        Add a disruption rule at the given layer
+        Add a filter at the given protocol layer for a given priority
         :param layer: The layer number to insert this rule at (must be one of the options in L2Tunnel.PARSE_CAPABLE_LAYERS)
-        :param rule: A lambda that excepts this layer's header and payload, modifies them, and returns them modified
+        :param resolution_index: The index at which to insert the rule into the processing queue. Rules are resolved in
+                                 the order of their resolution index, so a rule with a low resolution index is resolved
+                                 before a rule with a high resolution index. If the resolution index is greater than the
+                                 number of filters in the processing queue, then it is inserted at the end of the queue
+        :param protocol_filter: A callable that excepts this layer's header and payload, modifies them, and returns them modified
```

Comment:
excepts -> accepts - this is because you did not do Devra

Reply:
no you idiot, its because i misspelled expects.